src/Octoshift/Services/AdoApi.cs

```diff
@@ -700,6 +735,42 @@ public virtual async Task<bool> IsCallerOrgAdmin(string org)
         return await HasPermission(org, collectionSecurityNamespaceId, genericWritePermissionBitMaskValue);
     }
 
+    private string ExtractErrorMessage(string response, string dataProviderKey)
+    {
+        if (!response.HasValue())
+        {
+            return null;
+        }
+
+        try
+        {
+            var data = JObject.Parse(response);
+#pragma warning disable IDE0046 // Convert to conditional expression
+            if (data["dataProviders"] is not JObject dataProviders)
+            {
+                return null;
+            }
+#pragma warning restore IDE0046 // Convert to conditional expression
+
+            return dataProviders[dataProviderKey] is not JObject dataProvider ? null : (string)dataProvider["errorMessage"];
+        }
+        catch (JsonException ex)
+        {
```

Comment:
Don't swallow unexpected exceptions, if a JsonException, ArgumentException or InvalidOperationException occur let them bubble up and a higher level error handler will handle them properly.